### DISCUSSION TITLE
date given to the function to mutate the array turns to date -1

### DISCUSSION THREAD

**Discussion by Manu7864z** [neuefische/questions#539] [Web] [answered]
Hello there, 

i've got an issue in my project, this is the function: 

```jsx
const handleDateChange = (date) => {
  console.log(date); // Fri Mar 10 2021 00:00:00 GMT+0100 (Mitteleuropäische Normalzeit)
  setWeekDay(date.toISOString().slice(0, 10));
  setShowCalendar(!showCalendar);
  console.log(weekDay); // 2021-03-09 WHY?
};
```

as you can see i get a String and want to mutate the array into the format yyyy-mm-dd. 
this is working fine BUT if i choose the e.g. 10.0.2023 in the calendar and the function is called it mutates it but the output is always -1... so the date i get is 2023-03-09... i researched some hours and tried different things but this is the closest i can get.. Does anyone has an idea to fix it?

**Comment by PhilipZi** [comment, marked as answer]
can you show the entire code?

**Reply by Manu7864z** [reply]
hey philip i just found an solution to my problem.. 
this works for me now: 
` const handleDateChange = (date) => {
    console.log(date); // Fri Mar 10 2021 00:00:00 GMT+0100 (Mitteleuropäische Normalzeit)
    const getYear = date.toLocaleDateString("default", { year: "numeric" });
    const getMonth = date.toLocaleDateString("default", { month: "2-digit" });
    const getDay = date.toLocaleDateString("default", { day: "2-digit" });
    const newDate = `${getYear}-${getMonth}-${getDay}`;
    setWeekDay(newDate);`

**Reply by PhilipZi** [reply]
great :) well done
